eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -18,15 +18,29 @@ jobs:
   variables:
     displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
   pool:
-    name: Azure Pipelines
-    vmImage: ubuntu-18.04
+    name: azsdk-pool-mms-ubuntu-1804-general
+    vmImage: MMSUbuntu18.04
   displayName: Generate Job Matrix
   steps:
+    - checkout: none
+
+    - pwsh: |
+        git clone --no-checkout --filter=tree:0 git://github.com/$(Build.Repository.Name).git .
+        git sparse-checkout init
+        if ("${{ parameters.AdditionalParameters.ServiceDirectory }}") {
```

Comment:
Can `AdditionalParameters.ServiceDirectory` be a list?

Reply:
At the moment, ServiceDirectory as its passed in through the pipelines is a single value. In the java repo, we've started to add another parameter `TestResourceDirectories` to accommodate a list. My plan is to eventually support a list of directories (for this and ARM templates) across the repositories, so my thinking is when that change comes, we can make the updates here based on what it looks like.

Reply:
And yes, I think supporting an opt-in to selective checkout for ALL live test jobs is going to be nice to have, but there are more potential edge cases there. I'd like to start investigating that as a follow-on.

Reply:
Updated to include a list or a service directory default.